pkg/abstractions/function/function.go

```diff
@@ -397,6 +397,13 @@ func (fs *RunCFunctionService) FunctionSchedule(ctx context.Context, req *pb.Fun
 }
 
 func (fs *RunCFunctionService) listenForScheduledJobs() {
+	lock := common.NewRedisLock(fs.rdb)
+	if err := lock.Acquire(fs.ctx, functionScheduledJobsLock, common.RedisLockOptions{TtlS: 10, Retries: 3}); err != nil {
```

Comment:
I think the lock should be on the message handling itself, not the listening portion

Reply:
Updated.